src/lib/protocols/tftp.c

```diff
@@ -98,7 +98,8 @@ void ndpi_search_tftp(struct ndpi_detection_module_struct
           {
             ndpi_set_risk(ndpi_struct, flow, NDPI_MALFORMED_PACKET, "Invalid TFTP RR/WR header: Source/Destination file missing");
           } else {
-            memcpy(flow->protos.tftp.filename, &packet->payload[2], ndpi_min(filename_len, sizeof(flow->protos.tftp.filename) - 1));
+            filename_len = ndpi_min(filename_len, sizeof(flow->protos.tftp.filename) - 1);
+            memcpy(flow->protos.tftp.filename, &packet->payload[2], filename_len);
```

Comment:
I am at a loss. You basically split the statement into two and that fixes an uninitialized memory access issue? What do I miss here?

Reply:
got it..